src/WebJobs.Extensions.DurableTask/Listener/DurableTaskTargetScaler.cs

```diff
@@ -38,46 +39,55 @@ public DurableTaskTargetScaler(string functionId, DurableTaskMetricsProvider met
-            var orchestratorWorkers = Math.Min(activeControlQueues, upperBoundControlWorkers);
-
-            int numWorkersToRequest = (int)Math.Max(activityWorkers, orchestratorWorkers);
-            this.scaleResult.TargetWorkerCount = numWorkersToRequest;
-
-            // When running on ScaleController V3, ILogger logs are forwarded to the ScaleController's Kusto table.
-            // This works because this code does not execute in the Functions Host process, but in the ScaleController process,
-            // and the ScaleController is injecting it's own custom ILogger implementation that forwards logs to Kusto.
-            var scaleControllerLog = $"Target worker count for {this.functionId}: {numWorkersToRequest}. " +
-                $"Metrics used: workItemQueueLength={workItemQueueLength}. controlQueueLengths={serializedControlQueueLengths}. " +
-                $"maxConcurrentOrchestrators={this.MaxConcurrentOrchestrators}. maxConcurrentActivities={this.MaxConcurrentActivities}";
-
-            // target worker count should never be negative
-            if (numWorkersToRequest < 0)
+            DurableTaskTriggerMetrics? metrics = null;
+            try
             {
-                scaleControllerLog = "Tried to request a negative worker count." + scaleControllerLog;
-                this.logger.LogError(scaleControllerLog);
-
-                // Throw exception so ScaleController can handle the error.
-                throw new Exception(scaleControllerLog);
+                metrics = await this.metricsProvider.GetMetricsAsync();
+
+                // compute activityWorkers: the number of workers we need to process all activity messages
+                var workItemQueueLength = metrics.WorkItemQueueLength;
+                double activityWorkers = Math.Ceiling(workItemQueueLength / (double)this.MaxConcurrentActivities);
+
+                var serializedControlQueueLengths = metrics.ControlQueueLengths;
+                var controlQueueLengths = JsonConvert.DeserializeObject<IReadOnlyList<int>>(serializedControlQueueLengths);
+
+                var controlQueueMessages = controlQueueLengths.Sum();
+                var activeControlQueues = controlQueueLengths.Count(x => x > 0);
+
+                // compute orchestratorWorkers: the number of workers we need to process all orchestrator messages.
+                // We bound this result to be no larger than the partition count
+                var upperBoundControlWorkers = Math.Ceiling(controlQueueMessages / (double)this.MaxConcurrentOrchestrators);
+                var orchestratorWorkers = Math.Min(activeControlQueues, upperBoundControlWorkers);
+
+                int numWorkersToRequest = (int)Math.Max(activityWorkers, orchestratorWorkers);
+                this.scaleResult.TargetWorkerCount = numWorkersToRequest;
+
+                // When running on ScaleController V3, ILogger logs are forwarded to Kusto.
```

Comment:
Did you mean to change these comments? I recalled you added this extra detail in the previous PR but now it seems you're reverting it back.

Reply:
Ah I didn't meant to revert these changes, it was just a bad merge. I'm adding these back.